### PR TITLE
Add download (redirect) endpoint for replays

### LINKED ISSUE #295: Add download endpoints for vault
For counting the number of downloads we are currently relying on some ancient php code.

Basically the API takes the request, increases the downloadcount for this map/mod/replay and redirects to the content-server url.

### PR DESCRIPTION
**Pull request by Brutus5000**
partially fixes #295

### CODE REVIEW

File: src/main/java/com/faforever/api/download/DownloadController.java [hunk 1]
```diff
@@ -0,0 +1,23 @@
...
+@Controller
+@RequestMapping(path = "/download")
+@RequiredArgsConstructor
+public class DownloadController {
+  private final DownloadService downloadService;
+
+  @GetMapping("/replay/{id}")
+  public void downloadReplay(HttpServletResponse httpServletResponse,
+                             @PathVariable("id") int replayId) {
+    httpServletResponse.setHeader("Location", downloadService.getReplayDownloadUrl(replayId));
```
Comment (bukajsytlos) [inline src/main/java/com/faforever/api/download/DownloadController.java]: I would guess there is some constant for this header name

File: src/main/java/com/faforever/api/download/DownloadController.java [hunk 2]
```diff
@@ -0,0 +1,23 @@
...
+    httpServletResponse.setStatus(302);
```
Comment (bukajsytlos) [inline src/main/java/com/faforever/api/download/DownloadController.java]: and for status code too

File: src/main/java/com/faforever/api/download/DownloadController.java [hunk 3]
```diff
@@ -0,0 +1,23 @@
...
+import org.springframework.web.bind.annotation.GetMapping;
+import org.springframework.web.bind.annotation.PathVariable;
+import org.springframework.web.bind.annotation.RequestMapping;
+
+import javax.servlet.http.HttpServletResponse;
+
+@Controller
+@RequestMapping(path = "/download")
+@RequiredArgsConstructor
+public class DownloadController {
```
Comment (bukajsytlos) [inline src/main/java/com/faforever/api/download/DownloadController.java]: I would reconsider using ReplayController and url to be /replays/{id}/download

File: src/main/java/com/faforever/api/download/DownloadService.java [hunk 4]
```diff
@@ -0,0 +1,32 @@
+package com.faforever.api.download;
+
+import com.faforever.api.config.FafApiProperties;
+import lombok.RequiredArgsConstructor;
+import org.springframework.stereotype.Service;
+import org.springframework.util.Assert;
+
+@Service
+@RequiredArgsConstructor
+public class DownloadService {
```
Comment (bukajsytlos) [inline src/main/java/com/faforever/api/download/DownloadService.java]: and ReplayService

**Review by bukajsytlos — CHANGES_REQUESTED** [review]

File: src/main/java/com/faforever/api/game/GameController.java [hunk 5]
```diff
@@ -0,0 +1,25 @@
+package com.faforever.api.game;
+
+import lombok.RequiredArgsConstructor;
+import org.eclipse.jetty.http.HttpStatus;
```
Comment (bukajsytlos) [inline src/main/java/com/faforever/api/game/GameController.java]: just nitpick :)  org.springframework.http.HttpStatus#FOUND

File: src/main/java/com/faforever/api/game/GameService.java [hunk 6]
```diff
@@ -0,0 +1,34 @@
+package com.faforever.api.game;arengorman@gmail.com
+eugenevkopylov@gmail.com
```
Comment (bukajsytlos) [inline src/main/java/com/faforever/api/game/GameService.java]: hmmm?

**Review by bukajsytlos — CHANGES_REQUESTED** [review]

**Review by bukajsytlos — DISMISSED** [review]

File: src/main/java/com/faforever/api/game/GameService.java (hunk 6 above)
Comment (Brutus5000) [reply src/main/java/com/faforever/api/game/GameService.java]: already removed xD whatever copy&paste shit happened there

**Pull request merged**